presto-singlestore/src/main/java/com/facebook/presto/plugin/singlestore/SingleStoreClient.java

```diff
@@ -144,21 +149,36 @@ protected String toSqlType(Type type)
             if (varcharType.isUnbounded()) {
                 return "longtext";
             }
-            if (varcharType.getLengthSafe() <= 255) {
-                return "tinytext";
+            if (varcharType.getLengthSafe() <= 21844) {
+                // 21844 is the maximum length a singlestore varchar supports.
+                return super.toSqlType(type);
             }
-            if (varcharType.getLengthSafe() <= 65535) {
-                return "text";
-            }
-            if (varcharType.getLengthSafe() <= 16777215) {
+            if (varcharType.getLengthSafe() <= 5592405) { // 16MB
                 return "mediumtext";
             }
-            return "longtext";
+            if (varcharType.getLengthSafe() <= 1431655765) { // 100MB to 1GB
```

Comment:
The single store documentation [says](https://docs.singlestore.com/cloud/reference/sql-reference/data-types/string-types/) (emphasis mine) -  
>Note : There are several variants of the TEXT type, listed below. Aside from their maximum lengths, there is no practical difference between these types in SingleStore, **including performance or memory efficiency**. Further, TEXT , LONGTEXT, and VARCHAR are identical in functionality other than the maximum lengths. TEXT types are implemented solely for backward compatibility with MySQL syntax.

Since this method is primarily used when we create a temp table for writing data CREATE TABLE operation, can we simply use `LONGTEXT` for all VARCHAR types and remove these if-blocks ?